### PR TITLE
Chore/fixes n stuff

### PR DESCRIPTION
**Pull request by thestonefox**

### CODE REVIEW

File: Scripts/Process/MomentProcess.cs [hunk 1]
```diff
@@ -17,5 +17,9 @@ public sealed class MomentProcess : MonoBehaviour
         /// </summary>
         [Tooltip("Only run the process if the process is on an active GameObject and the Component is enabled.")]
         public bool onlyProcessOnActiveAndEnabled = true;
+
+        private void OnEnable()
```
Comment (bddckr) [inline Scripts/Process/MomentProcess.cs]: Let's add a comment here for when we wonder about this in the future:

```c#
// This empty implementation tells Unity to draw the enabled checkbox for this component, allowing to disable the component at edit-time.
```

File: Scripts/Tracking/Velocity/ApplyVelocity.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    /// <summary>
+    /// Applies the velocity data from the given <see cref="VelocityTracker"/> to the given <see cref="Rigidbody"/>.
+    /// </summary>
+    public class ApplyVelocity : MonoBehaviour
+    {
+        /// <summary>
+        /// The <see cref="Rigidbody"/> to apply the velocity data to.
+        /// </summary>
+        [Tooltip("The Rigidbody to apply the velocity data to.")]
+        public Rigidbody applyTo;
```
Comment (bddckr) [inline Scripts/Tracking/Velocity/ApplyVelocity.cs]: My suggestion is to call this `target`.

File: Scripts/Tracking/Velocity/ApplyVelocity.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        /// <summary>
+        /// The <see cref="VelocityTracker "/> to receive the velocity data from.
+        /// </summary>
+        [Tooltip("The VelocityTracker to receive the velocity data from.")]
+        public VelocityTracker velocityData;
```
Comment (bddckr) [inline Scripts/Tracking/Velocity/ApplyVelocity.cs]: This is not necessarily `data`, instead I suggest `source`. We don't need to talk about `velocity` in the member name here as it's implied by the component type we're in and the member's type.

File: Scripts/Tracking/Collision/CollisionList.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            protected set;
+        } = new List<CollisionNotifier.EventData>();
+
+        /// <summary>
+        /// Determines if the collection is empty.
+        /// </summary>
+        /// <returns><see langword="true"/> if the collection is empty.</returns>
+        public virtual bool IsEmpty()
+        {
+            return (Elements.Count == 0);
```
Comment (bddckr) [inline Scripts/Tracking/Collision/CollisionList.cs]: Do we really need to add a method for this? I'd rather have everyone type this out than introduce another method which allows to be overridden and thus break any usage of the list as the count and this empty check can get out of sync.

tl;dr: I don't see this saving us a lot so I wonder if it's worth removing it for the sake of ensuring the usage of the list is sound.

File: Scripts/Tracking/Collision/CollisionList.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            }
+            EmitChanged();
+        }
+
+        /// <summary>
+        /// Emits the Changed event.
+        /// </summary>
+        public virtual void EmitChanged()
+        {
+            Changed?.Invoke(Elements);
```
Comment (bddckr) [inline Scripts/Tracking/Collision/CollisionList.cs]: Same as above: I suggest to inline this method into its usages and remove it.

File: Scripts/Tracking/Collision/ListProcess/BaseListProcess.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        /// </summary>
+        [Tooltip("Executes any List Action components found on the collider.")]
+        public bool executeColliderActions = false;
+
+        /// <summary>
+        /// Emitted when the <see cref="CollisionList"/> collection has been processed.
+        /// </summary>
+        public CollisionList.UnityEvent Processed = new CollisionList.UnityEvent();
+
+        protected List<CollisionNotifier.EventData> processedList = new List<CollisionNotifier.EventData>();
```
Comment (bddckr) [inline Scripts/Tracking/Collision/ListProcess/BaseListProcess.cs]: Is this `previousList`? It seems like not all subclasses use it - should this instead be a thing in the subclasses that need it?

File: Scripts/Tracking/Collision/ListProcess/Action/BaseListAction.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    /// The base for providing an action that can be executed from a give collision.
+    /// </summary>
+    public abstract class BaseListAction : MonoBehaviour
+    {
+        /// <summary>
+        /// Executes an action on the given collision collection with the given state.
+        /// </summary>
+        /// <param name="collisionList">The collision collection at the time the action was executed.</param>
+        /// <param name="state">The state for the action.</param>
+        public abstract void Execute(List<CollisionNotifier.EventData> collisionList, bool state);
```
Comment (bddckr) [inline Scripts/Tracking/Collision/ListProcess/Action/BaseListAction.cs]: This is hardcoded to take a `bool`, what if I have a List Action that wants something else?
